foreman/data_refinery_foreman/foreman/test_end_to_end.py

```diff
@@ -168,7 +169,19 @@ def test_smasher_job(self):
             purge_experiment(accession_code)
 
         prepare_computed_files()
-        pyrefinebio.create_token(agree_to_terms=True, save_token=False)
+
+        # The API sometimes takes a bit to come back up.
+        stop = False
+        start_time = timezone.now()
+        while not stop:
+            try:
+                pyrefinebio.create_token(agree_to_terms=True, save_token=False)
+                stop = True
+            except pyrefinebio.ServerError:
+                if timezone.now() - start_time > timedelta(minutes=15):
+                    stop = True
```

Comment:
Should we fail the test here instead of breaking out of the loop? It seems like this would cause the test to fail later for a less-clear reason than if we raised an `AssertionError("Server not up after 15 minutes")`

Reply:
That's a good point!